editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts

```diff
@@ -594,11 +596,41 @@ export const editorStore: StateCreator<
 
     // Determine node path based on the node type
     const nodePath =
-      node.type === ComponentType.Answer
+      node.type === TYPES.Answer
         ? `nodes/${grandparent.id}/nodes/${parent.id}/edit`
         : `nodes/${parent.id}/nodes/${node.id}/edit`;
 
     const urlPath = `/${teamSlug}/${flowSlug}${portalPath}/${nodePath}`;
     return urlPath;
   },
+
+  getFlowSchema: () => {
+    const { flow } = get();
+    if (!flow) return;
+
+    const nodes: Set<string> = new Set();
+    const options: Set<string> = new Set();
+
+    Object.entries(flow).map(([_id, node]) => {
+      if (node.data?.fn) {
+        // Exclude Filter fn value as not exposed to editors
+        if (node.data?.fn !== "flag") nodes.add(node.data.fn)
+      };
+  
+      // TODO align to (reuse?) data facets from search
```

Comment:
Great comment!

We could probably use `DATA_FACETS` here (filtered to strings only) and then use this to get the data values from each node.

However, I think this is actually a more complex solution - there's a very different use case here and we'll have just as many exceptions and clauses. This works really well for now - we just need to make sure that we keep data values for new components, this function, and `DATA_FACETS` in sync.

Reply:
Yep - `fn` only basically felt like right "first" go at this since other `DATA_FACETS` like FileUploadAndLabel individual file fns are getting "overwritten" by ODP Schema enum suggestions anyways ! So it's a good question of whether or not we actually always want this to be perfectly in sync !